src/main/java/it/gov/pagopa/payhub/activities/activity/classifications/IufClassificationActivityImpl.java

```diff
@@ -0,0 +1,89 @@
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+/**
+ * Interface for defining an activity to process payment reporting classifications based on IUF.
+ */
+@Slf4j
+@Lazy
+@Component
+public class IufClassificationActivityImpl implements IufClassificationActivity {
+    private final PaymentsReportingDao paymentsReportingDao;
+    private final TreasuryDao treasuryDao;
+    private final ClassifyDao classifyDao;
+
+    private static final String TES_NO_MATCH = "TES_NO_MATCH";
+
+    public IufClassificationActivityImpl(PaymentsReportingDao paymentsReportingDao, TreasuryDao treasuryDao, ClassifyDao classifyDao) {
+        this.paymentsReportingDao = paymentsReportingDao;
+        this.treasuryDao = treasuryDao;
+        this.classifyDao = classifyDao;
+    }
+
+    /**
+     * Processes IUF classification based on the provided parameters.
+     *
+     * @param organizationId the unique identifier of the organization
+     * @param iuf            the unique identifier of the payment reporting flow (IUF)
+     * @return IufClassificationActivityResult containing PaymentsReportingDTO list and success flag
+     */
+    @Override
+    public IufClassificationActivityResult classify(Long organizationId, String iuf) {
+        if (organizationId==null || organizationId.equals(0L))
+            throw new PaymentsClassificationSaveException("organization id may be not null or zero");
+        if (iuf==null || iuf.isEmpty())
+            throw new PaymentsClassificationSaveException("iuf may be not null or blank");
+
+        IufClassificationActivityResult iufClassificationActivityResult = new IufClassificationActivityResult();
+        List<PaymentsReportingDTO> paymentsReportingDTOS = paymentsReportingDao.findByOrganizationIdFlowIdentifierCode(organizationId, iuf);
+
+        for (TreasuryDTO treasuryDTO : treasuryDao.searchByIuf(iuf)) {
+            try {
+                setAndSave(paymentsReportingDTOS.size(), treasuryDTO.getMygovFlussoTesoreriaId());
+            }
+            catch (PaymentsClassificationSaveException paymentsClassificationSaveException) {
+                log.error("Error saving classification");
+                throw new PaymentsClassificationSaveException("Error saving classification");
+            }
+        }
+        return iufClassificationActivityResult.toBuilder()
+                .paymentsReportingDTOS(paymentsReportingDTOS)
+                .success(true)
+                .build();
+    }
+
+    /**
+     *
+     * @param paymentsReportingSize size of the list of payments reporting found
+     * @param treasuryId primary key of Treasury
+     * @throws PaymentsClassificationSaveException exception
+     */
+    private void setAndSave(int paymentsReportingSize, Long treasuryId) throws PaymentsClassificationSaveException {
+        if (paymentsReportingSize==0 && treasuryId!=null) {
+            ClassifyDTO classifyDTO = ClassifyDTO.builder()
```

Comment:
we should save also other info, as the iuf and the organizationId as an example (at least the same info expected on the Clear activity)

Reply:
Added in save organizationId, treasuryId and iuf
@marcomatteuccieng is refactoring table

Reply:
moved outside:
   if (paymentsReportingSize==0 && treasuryId!=null) {
the second not needed and the first is only for verifify an empty list